zaza/openstack/charm_tests/cinder_backup/tests.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 Canonical Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Encapsulate cinder-backup testing."""
+import time
+import logging
+
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.ceph as ceph_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+
+
+class CinderBackupTest(test_utils.OpenStackBaseTest):
+    """Encapsulate Cinder Backup tests."""
+
+    RESOURCE_PREFIX = 'zaza-cinderbackuptests'
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Cinder Backup tests."""
+        super(CinderBackupTest, cls).setUpClass()
+        cls.cinder_client = openstack_utils.get_cinder_session_client(
+            cls.keystone_session)
+
+    @property
+    def services(self):
+        """Return a list services for Openstack Release."""
+        current_release = openstack_utils.get_os_release()
+        services = ['cinder-scheduler', 'cinder-volume']
+        if (current_release >=
+                openstack_utils.get_os_release('xenial_ocata')):
+            services.append('apache2')
+        else:
+            services.append('cinder-api')
+        return services
+
+    def test_100_volume_create_extend_delete(self):
+        """Test creating, extending a volume."""
+        vol_new = openstack_utils.create_volume(
+            self.cinder_client,
+            name='{}-100-vol'.format(self.RESOURCE_PREFIX),
+            size=1)
+        self.cinder_client.volumes.extend(
+            vol_new.id,
+            '2')
+        openstack_utils.resource_reaches_status(
+            self.cinder_client.volumes,
+            vol_new.id,
+            expected_status="available",
+            msg="Volume status wait")
+
+    def test_410_cinder_vol_create_backup_delete_restore_pool_inspect(self):
+        """Create, backup, delete, restore a ceph-backed cinder volume.
+
+        Create, backup, delete, restore a ceph-backed cinder volume, and
+        inspect ceph cinder pool object count as the volume is created
+        and deleted.
+        """
+        unit_name = 'ceph-mon/0'
```

Comment:
This assumption about the unit name may be wrong. I think it'd be safer to use `zaza.model.get_lead_unit_name`

Reply:
Will `zaza.model.get_lead_unit_name('ceph-mon')` work?

Reply:
Seems to:
```
>>> import zaza.model
>>> print(zaza.model.get_lead_unit_name('memcached'))
memcached/0
>>> print(type(zaza.model.get_lead_unit_name('memcached')))
<class 'str'>
```

Reply:
Great, tranks!